### PR TITLE
Fix popup window width calculation for Vim #86

### PR DESCRIPTION
**Pull request by brglng**

### CODE REVIEW

**Comment by liuchengxu** [comment]
@tuberry You can see if this PR helps.

File: autoload/which_key/window.vim [hunk 1]
```diff
@@ -63,10 +63,10 @@ function! s:show_popup(rows) abort
   let offset = s:floating_win_col_offset()
   if g:which_key_floating_relative_win
     let col = offset + win_screenpos(g:which_key_origin_winid)[1]
-    let maxwidth = winwidth(g:which_key_origin_winid) - offset - 1
+    let maxwidth = winwidth(g:which_key_origin_winid) - offset + 1
```
Comment (tuberry) [inline autoload/which_key/window.vim]: let maxwidth = winwidth(g:which_key_origin_winid) - offset

File: autoload/which_key/window.vim [hunk 2]
```diff
@@ -63,10 +63,10 @@ function! s:show_popup(rows) abort
...
   else
     let col = offset
-    let maxwidth = &columns - offset - 1
+    let maxwidth = &columns - offset + 1
```
Comment (tuberry) [inline autoload/which_key/window.vim]: let maxwidth = &columns - offset

Comment (brglng) [reply autoload/which_key/window.vim]: > ```
>  let maxwidth = &columns - offset
> ```

It looks like Vim's popup uses column number starting from 1. Maybe we should use:
```
function! s:floating_win_col_offset() abort
  return (&number ? strlen(line('$')) : 0) + (&signcolumn ==# 'yes' ? 2: 0)
endfunction
...
let col = offset + 1
...
let maxwidth = winwidth(g:which_key_origin_winid) - offset
...
let maxwidth = &columns - offset
```
?

**Pull request merged**